test/unittest/base/test_datalayer.py

```diff
@@ -47,7 +47,7 @@ def pre_create(self, db):
     def on_load(self, db):
         self.is_on_load = True
 
-    def schedule_jobs(self, db, dependencies):
+    def schedule_jobs(self, db, dependencies, distributed=False):
```

Comment:
What's this flag:`distributed=False`? Why necessary?

Reply:
when someone does, 
inside db.add earlier it was 
```
object.schedule_jobs(db, dependencies)
```
since no distributed was sent it was had distributed as False in Listener 

I changed in 
```
object.schedule_jobs(db, dependencies, distributed=self.distributed)
```

now TestComponent class `schedule_jobs`
also need to take  distributed param as we are passing it in schedule_jobs